src/github3/repos/repo.py

```diff
@@ -517,7 +557,109 @@ def contributors(self, anon=False, number=-1, etag=None):
             params = {"anon": "true"}
         return self._iter(int(number), url, users.Contributor, params, etag)
 
-    @requires_auth
+    @decorators.requires_app_installation_auth
+    def create_check_run(
+        self,
+        name,
+        head_sha,
+        details_url=None,
+        external_id=None,
+        started_at=None,
+        status=None,
+        conclusion=None,
+        completed_at=None,
+        output=None,
+        actions=None,
+    ):
+        """Create a check run object on a commit
+
+        .. versionadded:: 1.3.0
+
+        :param str name:
+            (required), The name of the check
+        :param str head_sha:
+            (required), The SHA of the commit
+        :param str details_url:
+            (optional), The URL of the integrator's site that has the full
+            details of the check
+        :param str external_id:
+            (optional), A reference for the run on the integrator's system
+        :param str started_at:
+            (optional), ISO 8601 time format: YYYY-MM-DDTHH:MM:SSZ
+        :param str status:
+            (optional), ('queued', 'in_progress', 'completed')
+        :param str conclusion:
+            (optional), Required if you provide 'completed_at', or a
+            'status' of 'completed'. The final conclusion of the check.
+            ('success', 'failure', 'neutral', 'cancelled', 'timed_out',
+            'action_required')
+        :param str completed_at:
+            (optional), Required if you provide 'conclusion'. ISO 8601 time
+            format: YYYY-MM-DDTHH:MM:SSZ
+        :param dict output:
+            (optional), key-value pairs representing the output. Format:
+            {'title': 'string', 'summary': 'text, can be markdown', 'text':
+            'text, can be markdown', 'annotations': [{}], 'images': [{}]}
+        :param array actions:
+            (optional), array of action objects. Object format is:
+            {'label': 'text', 'description', 'text', 'identifier', 'text'}
```

Comment:
Same comment as above. This needs clarification

Reply:
Done.